File: examples/governance_static_go/domain/regulatory_framework_repository.go
package domain

type RegulatoryFrameworkRepository interface {
	Find(id string) (*RegulatoryFramework, error)
	Save(RegulatoryFramework *RegulatoryFramework) error
	All() ([]*RegulatoryFramework, error)
	Delete(id string) error
	Count() (int, error)
}

File: examples/governance/compliance_domain/lib/compliance_domain/ports/regulatory_framework_repository.rb
module ComplianceDomain
  module Ports
    module RegulatoryFrameworkRepository
      def find(id)
        raise NotImplementedError, "#{self.class}#find not implemented"
      end

      def save(regulatory_framework)
        raise NotImplementedError, "#{self.class}#save not implemented"
      end

      def delete(id)
        raise NotImplementedError, "#{self.class}#delete not implemented"
      end
    end
  end
end
